internal/buildscripts/packaging/choco/splunk-otel-collector/tools/chocolateyinstall.ps1

```diff
@@ -149,24 +149,43 @@ $packageArgs = @{
     validExitCodes = @(0)
 }
 
-Install-ChocolateyInstallPackage @packageArgs
+$use_msi_properties = [Version]$env:chocolateyPackageVersion -ge [Version]"0.99.0"
+if ($use_msi_properties) {
+    $msi_properties_args = ""
+    foreach ($entry in $env_vars.GetEnumerator()) {
+        $msi_properties_args += " $($entry.Key)=`"$($entry.Value)`""
+    }
 
-if ($MODE -eq "agent" -or !$MODE) {
-    $config_path = "$program_data_path\agent_config.yaml"
-    if (-NOT (Test-Path -Path "$config_path")) {
-        write-host "Copying agent_config.yaml to $config_path"
-        Copy-Item "$installation_path\agent_config.yaml" "$config_path"
+    if ($MODE) {
+        $msi_properties_args += " SPLUNK_SETUP_COLLECTOR_MODE=`"$MODE`""
     }
+    Write-Host "Using MSI properties: $msi_properties_args"
+    Write-Host "SilentArgs: $($packageArgs["silentArgs"])"
+    $packageArgs["silentArgs"] += $msi_properties_args
+    Write-Host "SilentArgs: $($packageArgs["silentArgs"])"
 }
-elseif ($MODE -eq "gateway"){
-    $config_path = "$program_data_path\gateway_config.yaml"
-    if (-NOT (Test-Path -Path "$config_path")) {
-        write-host "Copying gateway_config.yaml to $config_path"
-        Copy-Item "$installation_path\gateway_config.yaml" "$config_path"
+
+Install-ChocolateyInstallPackage @packageArgs
+
+if (!$use_msi_properties) {
+    if ($MODE -eq "agent" -or !$MODE) {
+        $config_path = "$program_data_path\agent_config.yaml"
+        if (-NOT (Test-Path -Path "$config_path")) {
+            write-host "Copying agent_config.yaml to $config_path"
+            Copy-Item "$installation_path\agent_config.yaml" "$config_path"
+        }
+    }
+    elseif ($MODE -eq "gateway"){
+        $config_path = "$program_data_path\gateway_config.yaml"
+        if (-NOT (Test-Path -Path "$config_path")) {
+            write-host "Copying gateway_config.yaml to $config_path"
+            Copy-Item "$installation_path\gateway_config.yaml" "$config_path"
+        }
     }
+
+    $env_vars["SPLUNK_CONFIG"] = "$config_path"

```

Comment:
Shouldn't this still be outside so it's set regardless of `$use_msi_properties`?

Reply:
No, the MSI handles that automatically.